### DISCUSSION TITLE
Can I custom the client-side fetch implement?

### DISCUSSION THREAD

**Discussion by lili21** [remix-run/remix#3430] [Q&A] [answered]
my web app run inside a Native App's Webview. For some reason, I need to rely the Native App to call the backend API. 
is there anyway I can do that?

**Comment by sergiodxa** [comment, marked as answer]
You can use [resource routes](https://remix.run/docs/en/v1/guides/resource-routes) to create an [API endpoint](https://remix.run/docs/en/v1/guides/api-routes).

Or do you want to change how Remix does the request internally?

**Reply by lili21** [reply]
Yeah, the second one.

**Reply by sergiodxa** [reply]
Client-side Remix uses fetch, you could do something like this probably
```ts
let originalFetch = window.fetch
window.fetch = async function customFetch(url, init) {
  // do things inside this file and call originalFetch at some moment
  let response = await originalFetch(url, init)
  // you could even do things with the response here
  return response
}
```
I'm not sure if that solves what you want to do

**Reply by lili21** [reply]
Thanks, that should work. I'll try it later.
